testing/citest/tests/gcs_pubsub_gae_test.py

```diff
@@ -0,0 +1,453 @@
+        status_path=executions_path
+      ),
+      contract=builder.build())
+
+  def delete_load_balancer(self):
+    bindings = self.bindings
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            'type': 'deleteLoadBalancer',
+            'cloudProvider': 'appengine',
+            'loadBalancerName': self.__lb_name,
+            'account': bindings['SPINNAKER_APPENGINE_ACCOUNT'],
+            'credentials': bindings['SPINNAKER_APPENGINE_ACCOUNT'],
+            'user': '[anonymous]'
+        }],
+        description='Delete Load Balancer: {0} in {1}'.format(
+            self.__lb_name,
+            bindings['SPINNAKER_APPENGINE_ACCOUNT']),
+        application=self.TEST_APP)
+
+    builder = gcp.GcpContractBuilder(self.appengine_observer)
+    (builder.new_clause_builder('Service Deleted', retryable_for_secs=30)
+     .inspect_resource('apps.services',
+                       self.__lb_name,
+                       appsId=self.__gcp_project)
+     .EXPECT(
+         ov_factory.error_list_contains(gcp.HttpErrorPredicate(http_code=404))))
+
+    return st.OperationContract(
+        self.new_post_operation(
+            title='delete_load_balancer', data=payload, path='tasks'),
+        contract=builder.build())
+
+  def delete_deploy_pipeline(self, pipeline_id):
+    payload = self.agent.make_json_payload_from_kwargs(id=pipeline_id)
+    path = os.path.join('pipelines', self.TEST_APP, pipeline_id)
+
+    builder = st.HttpContractBuilder(self.agent)
+    (builder.new_clause_builder('Has Pipeline',
+                                retryable_for_secs=5)
+     .get_url_path(
+       'applications/{app}/pipelineConfigs'.format(app=self.TEST_APP))
+     .excludes_path_value('name', pipeline_id))
+
+    return st.OperationContract(
+      self.new_delete_operation(
+        title='delete_bake_pipeline', data=payload, path=path,
+        status_class=st.SynchronousHttpOperationStatus),
+      contract=builder.build())
+
+class GcsPubsubGaeTest(st.AgentTestCase):
+  @classmethod
+  def setUpClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    scenario = runner.get_shared_data(GcsPubsubGaeTestScenario)
+    bindings = scenario.bindings
+
+    branch = bindings['BRANCH']
+    git_repo = bindings['GIT_REPO_URL']
+
+    gcs_path = 'gs://{bucket}'.format(bucket=scenario.bucket)
+    topic = '{}-topic'.format(bindings['TEST_APP'])
+    subscription = '{}-subscription'.format(bindings['TEST_APP'])
+
+    # App to tar and upload to GCS.
+    command = 'git clone {repo} -b {branch} {dir}'.format(
+      repo=git_repo, branch=branch, dir=scenario.temp)
+    logging.info('Fetching %s', git_repo)
+    subprocess.Popen(command, stderr=sys.stderr, shell=True).wait()
+
+
+  @classmethod
+  def tearDownClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    scenario = runner.get_shared_data(GcsPubsubGaeTestScenario)
+    bindings = scenario.bindings
+    shutil.rmtree(scenario.temp)
```

Comment:
where did scenario.temp come from?

Reply:
It's a temp directory to clone the test appengine app -- we use it in multiple steps of the test so it needs to live the duration of the scenario.

Reply:
Why is it removed here? Ownership of the directory seems off. What if multiple fixtures are accessing it?

Reply:
The temp directory is created and managed in this fixture only.